hyperframe/frame.py

```diff
@@ -381,20 +391,21 @@ def __init__(self, stream_id=0, settings=None, **kwargs):
         self.settings = settings or {}
 
     def serialize_body(self):
-        settings = [struct.pack("!HL", setting & 0xFF, value)
-                    for setting, value in self.settings.items()]
-        return b''.join(settings)
+        return b''.join([_STRUCT_HL.pack(setting & 0xFF, value)
+                         for setting, value in self.settings.items()])
 
     def parse_body(self, data):
+        number_of_settings = 0
         for i in range(0, len(data), 6):
             try:
-                name, value = struct.unpack("!HL", data[i:i+6])
+                name, value = _STRUCT_HL.unpack(data[i:i+6])
             except struct.error:
                 raise InvalidFrameError("Invalid SETTINGS body")
 
             self.settings[name] = value
+            number_of_settings += 1
 
-        self.body_len = len(data)
+        self.body_len = number_of_settings * 6
```

Comment:
You can skip a multiplication here by, instead of counting `number_of_settings`, having `body_len` as the counter and incrementing it by 6 for each entry.